code/src/test_usart.cpp

```diff
@@ -0,0 +1,57 @@
+#include <test_usart.hpp>
+
+namespace r2d2 {
+
+    test_usart_c::test_usart_c(){}
+
+    void test_usart_c::enable(){
+        //Not needed in test implementation
+    }
+
+    void test_usart_c::disable(){
+        //Not needed in test implementation
+    }
+
+    bool test_usart_c::send(const uint8_t c) {
+        return true;
+        //Not needed in test implementation
+    }
+
+    void test_usart_c::putc(char c) {
+        send(c);
+    }
+
+    uint8_t test_usart_c::receive() {
+        return receive_buffer.copy_and_pop();
+    }
+
+    bool test_usart_c::char_available() {
+        return !receive_buffer.empty();
+    }
+
+    char test_usart_c::getc() {
+        return receive();
+    }
+
+    unsigned int test_usart_c::available() {
+        return !receive_buffer.empty();
+    }
+
+    void test_usart_c::set_receive_string(const std::string &str) {
+        receive_buffer.clear();
+        for(char c : str){
+            set_receive_byte(c);
+        }
+    }
+
+    void test_usart_c::set_receive_bytes(const std::vector<uint8_t> &bytes){
+        receive_buffer.clear();
+        for(auto &byte : bytes){
+            set_receive_byte(byte);
+        }
+    }
+
+    void test_usart_c::set_receive_byte(const uint8_t byte){
```

Comment:
this function sounds like the byte i set here will be the only byte that will be received. Maybe a name change to something like ```add_receive_byte()``` is better here

Reply:
or a clear before the push is oke as well i guess.

Reply:
was changed in https://github.com/R2D2-2019/USART_library/pull/12/commits/85fdce96b4074228fcd84ff4c1db1e03da94d3f2